contracts/accountants/GenericAccountant.vy

```diff
@@ -50,59 +97,125 @@ MAX_SHARE: constant(uint256) = 7_500
+    assert default_management <= self._management_fee_threshold(), "exceeds management fee threshold"
+    assert default_performance <= self._performance_fee_threshold(), "exceeds performance fee threshold"
+
+    # Set initial addresses
+    self.fee_manager = fee_manager
+    self.fee_recipient = fee_recipient
+
+    # Set the default fee config
+    self.default_config = Fee({
+        management_fee: default_management,
+        performance_fee: default_performance,
+        refund_ratio: default_refund,
+        max_fee: default_max
+    })
+
+    log UpdateDefaultFeeConfig(self.default_config)
 
 
 @external
 def report(strategy: address, gain: uint256, loss: uint256) -> (uint256, uint256):
-    """ """
-    total_refunds: uint256 = 0
-
-    # management_fee is charged in both profit and loss scenarios
+    """ 
+    @notice To be called by a vault during the process_report in which the accountant
+        will charge fees based on the gain or loss the strategy is reporting.
+    @dev Can only be called by a vault that has been added to this accountant.
+        Will default to the default_config for all amounts unless a custom config
+        has been set for a specific strategy.
+    @param strategy The strategy that is reporting.
+    @param gain The profit the strategy is reporting if any.
+    @param loss The loss the strategy is reporting if any.
+    """
+    # Make sure this is a valid vault.
+    assert self.vaults[msg.sender], "!authorized"
+
+    # Retrieve the strategies params from the vault.
     strategy_params: StrategyParams = IVault(msg.sender).strategies(strategy)
-    fee: Fee = self.fees[strategy]
-    duration: uint256 = block.timestamp - strategy_params.last_report
+    # Expected behavior is to use the default config.
+    fee: Fee = self.default_config
+
+    # Use custom fees if applicable.
+    if self.custom[strategy]:
+        fee = self.fees[strategy]
+
     total_fees: uint256 = 0
+    total_refunds: uint256 = 0
 
+    # Charge management fees no matter gain or loss.
     if fee.management_fee > 0:
-        #management_fee
+        # Time since last harvest.
+        duration: uint256 = block.timestamp - strategy_params.last_report
+        # management_fee is an annual amount, so charge based on the time passed.
         total_fees = (
             strategy_params.current_debt
             * duration
-            * fee.management_fee
+            * convert(fee.management_fee, uint256)
             / MAX_BPS
             / SECS_PER_YEAR
         )
 
+    # Only charge performance fees if there is a gain.
     if gain > 0:
-        total_fees += (gain * fee.performance_fee) / MAX_BPS
-        # ensure fee does not exceed more than 75% of gain
-        maximum_fee: uint256 = (gain * MAX_SHARE) / MAX_BPS
-        # test with min?
-        if total_fees > maximum_fee:
-            return (maximum_fee, 0)
+        total_fees += (gain * convert(fee.performance_fee, uint256)) / MAX_BPS
     else:
```

Comment:
Yes, but theres no reason to be calling report if there is no gain or loss right? So your already wasting an extreme amount of gas by reporting nothing. Would rather save the minor gas for an extra if check for the correct usage instead of the extra gas spent for mistakes that shouldnt happen. And it doesnt cause anything bad to happen.